GAE/src/com/gallatinsystems/survey/dao/QuestionGroupDao.java

```diff
@@ -79,6 +85,16 @@ public void deleteGroupsForSurvey(Long surveyId) throws IllegalDeletionException
         if (groups != null) {
             int i = 1;
             for (QuestionGroup group : groups) {
+                List<Translation> translations = translationDao.findTranslations(
+                        group.getKey().getId(), Translation.ParentType.QUESTION_GROUP_DESC,
+                        Translation.ParentType.QUESTION_GROUP_NAME);
+                HashMap<String, Translation> translationMap = new HashMap<>();
+                if (translations != null) {
+                    for (Translation t: translations) {
+                        translationMap.put(t.getLanguageCode(), t);
```

Comment:
Wont it happen here that if we have a list of translation for both the group description and the group name that one will be overwritten for the same language?

Reply:
you are right, after checking the Translations, it seem group description does not exist but I removed it from the query just in case